src/java/com/sarality/app/view/list/ComponentComparator.java

```diff
@@ -0,0 +1,11 @@
+package com.sarality.app.view.list;
```

Comment:
java comparator compares two items of the same type.  whereas we need the search string to be compared with the data object (contact data in this case).